### PR TITLE
add waterfall chart to gallery

### PR DESCRIPTION
**Pull request by RossKen**
### Type of PR

- [ ] BUG
- [ ] FEAT
- [ ] MAINT
- [X] DOC

### PR Checklist

- [ ] Added documentation for changes
- [ ] Added feature to example notebooks or tutorial (if appropriate)
- [ ] Added tests (if appropriate)
- [ ] Made changes based off the latest version of Splink
- [ ] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint.html)

### CODE REVIEW

File: docs/charts/waterfall_chart.ipynb [hunk 1]
```diff
@@ -173,15 +170,54 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### How to interpret the chart\n",
+    "\n",
+    "The first bar (labelled \"Prior\") is the match weight if no additional knowledge of features is taken into account, and can be thought of as similar to the y-intercept in a simple regression.\n",
+    "\n",
+    "Each subsequent bar shows the match weight for a comparison. These bars can be positive or negative depending on whether the given comparison gives positive or negative evidence for the two records being a match.\n",
+    "\n",
+    "Additional bars are added for comparisons with term frequency adjustments. For example, the chart above has term frequency adjustments for `first_name` so there is an extra `tf_first_name` bar showing the impact how the frequency of a given name impacts the amount of evidence for the two records being a match.\n",
```
Comment (ThomasHepworth) [inline docs/charts/waterfall_chart.ipynb:210]:
```suggestion
    "Additional bars are added for comparisons with term frequency adjustments. For example, the chart above has term frequency adjustments for `first_name` so there is an extra `tf_first_name` bar showing how the frequency of a given name impacts the amount of evidence for the two records being a match.\n",
```

File: docs/charts/waterfall_chart.ipynb [hunk 2]
```diff
@@ -190,7 +226,18 @@
    "metadata": {},
    "source": [
     "### Actions to take as a result of the chart\n",
-    "\n"
+    "\n",
+    "This chart is useful for spot checking pairs of records to see if the Splink model is behaving as expected.\n",
+    "\n",
+    "If a pair of records look like the are incorrectly being assigned as a match/non-match, it is a sign that the Splink model is not working optimally. If this is the case, it is worth revisiting the model training step. Some common scenarios include:\n",
```
Comment (ThomasHepworth) [inline docs/charts/waterfall_chart.ipynb:232]:
```suggestion
    "If a pair of records look like they are incorrectly being assigned as a match/non-match, it is a sign that the Splink model is not working optimally. If this is the case, it is worth revisiting the model training step. Some common scenarios include:\n",
```

Comment (ThomasHepworth) [reply docs/charts/waterfall_chart.ipynb:232]: Newline `Some common scenarios include`?

File: docs/charts/waterfall_chart.ipynb [hunk 3]
```diff
@@ -190,7 +226,18 @@
...
+    "\n",
+    "- If the \"Prior\" match weight is too small or large compared to the match weight provided by the comparisons, try some different determininstic rules and recall inputs to the [`estimate_probability_two_records_match` function](../linker.md#splink.linker.Linker.estimate_probability_two_random_records_match).\n",
+    "\n",
+    "- If you are working with a model with term frequency adjustments and want to dig deeper into the impact of term frequency on the model as a whole (i.e. not just for a single pairwise comparison), check out the [`tf_adjustment_chart`](./tf_adjustment_chart.ipynb).\n"
```
Comment (ThomasHepworth) [inline docs/charts/waterfall_chart.ipynb:240]: If we add a "model refinement" guide somewhere into the docs at some stage, it likely makes sense to add a link to here.

**Review by ThomasHepworth — APPROVED** [review]
Minor comments to be addressed, but feel free to merge once you've addressed them.

**Comment by RossKen** [comment]
Thanks @ThomasHepworth! Changes made so going to merge

**Pull request merged**